### PR TITLE
More show methods and tests for factorizations

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by andreasnoack — APPROVED** [review]

File: test/linalg/schur.jl [hunk 1]
```diff
@@ -38,6 +38,11 @@ aimg  = randn(n,n)/2
         @test AbstractArray(f) ≈ a
         @test_throws KeyError f[:A]
 
+        tstring = sprint(show,f[:T])
+        zstring = sprint(show,f[:Z])
+        vstring = sprint(show,f[:values])
+        @test sprint(show,f) == "$(typeof(f)) with factors T and Z:\n$tstring\n$zstring"*"and values:\n$vstring"
```
Comment (Sacha0) [inline test/linalg/schur.jl]: Why the string concatenation?

Comment (kshyatt) [inline test/linalg/schur.jl]: because otherwise, it was looking for `$zstringand` which doesn't exist

Comment (Sacha0) [inline test/linalg/schur.jl]: Perhaps `$(zstring)and` would be clearer?

File: base/linalg/schur.jl [hunk 2]
```diff
@@ -61,6 +61,15 @@ function getindex(F::Schur, d::Symbol)
     end
 end
 
+function show(io::IO, F::Schur)
+    println(io, "$(typeof(F)) with factors T and Z:")
+    show(io, F[:T])
+    println(io)
+    show(io, F[:Z])
+    println(io, "and values:")
```
Comment (tkelman) [inline base/linalg/schur.jl]: is this going to end up on the same line as the Z factor? possibly without any spacing between the factor array and the "and " ... ?

Comment (kshyatt) [inline base/linalg/schur.jl]: Yeah, good catch, fixing now.

**Review by Sacha0 — APPROVED** [review]

**Pull request merged**